assets/dev/js/editor/regions/panel/pages/elements/elements.js

```diff
@@ -160,6 +172,28 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
 		this.categoriesCollection = categoriesCollection;
 	},
 
+	shouldNotAddWidget( widget ) {
+		const isContainerActive = elementorCommon.config.experimentalFeatures.container;
+
+		return ! widget.show_in_panel || ( 'inner-section' === widget.name && isContainerActive );
+	},
+
+	deepMerge( originalObj, replacementObj ) {
+		const mergedObj = { ...originalObj };
+
+		for ( const key in replacementObj ) {
+			if ( replacementObj.hasOwnProperty( key ) ) {
+				if ( 'object' === typeof replacementObj[ key ] && replacementObj[ key ] !== null && originalObj.hasOwnProperty( key ) && 'object' === typeof originalObj[ key ] && originalObj[ key ] !== null ) {
```

Comment:
can this long condition be extracted to a variable that says what we are checking?

Reply:
Done